lib/saputils.pm

```diff
@@ -31,63 +32,159 @@ SAP Applications.
+                ($parameter, $value);
+            } @resources_parameters;
+            $topology{'Resource'}{$resource} = \%resource_parameter;
+        }
+
+        for my $host (@all_hosts) {
+
+            # New structure introduces key 'Site' to which some values are moved and have keys renamed
+            # or left defined but empty if it's not defined originally
+            my $sth_site = $script_topology{$host}->{'site'};
+            if (defined $sth_site) {
+                $topology{'Site'}{$sth_site}{'mns'} = $host;
+                if (defined $script_topology{$host}->{'op_mode'}) { $topology{'Site'}{$sth_site}{'opMode'} = $script_topology{$host}->{'op_mode'} }
+                if (defined $script_topology{$host}->{'srmode'}) { $topology{'Site'}{$sth_site}{'srMode'} = $script_topology{$host}->{'srmode'} }
+                if (defined $script_topology{$host}->{'sync_state'}) { $topology{'Site'}{$sth_site}{'srPoll'} = $script_topology{$host}->{'sync_state'} }
+
+                # Unfortunately, the new structure lack the key 'node_state' completely, so we need use the
+                # new key 'lss' which represents the state of the cluster '4' mean OK '1' means FAILED
+                if (defined $script_topology{$host}->{'node_state'}) {
+                    $topology{'Site'}{$sth_site}{'lss'} = ($script_topology{$host}->{'node_state'} eq 'online' or $script_topology{$host}->{'node_state'} =~ /[1-9]+/) ? '4' : '1';
+                }
+            }
+
+            # New structure rename key 'Hosts' to the 'Host' and also get keys renamed
+            # or left defined but empty if it's not defined originally
+            if (defined $script_topology{$host}->{'vhost'}) { $topology{'Host'}{$host}{'vhost'} = $script_topology{$host}->{'vhost'} }
+            if (defined $sth_site) { $topology{'Host'}{$host}{'site'} = $sth_site }
+            if (defined $script_topology{$host}->{'srah'}) { $topology{'Host'}{$host}{'srah'} = $script_topology{$host}->{'srah'} }
+            if (defined $script_topology{$host}->{'clone_state'}) { $topology{'Host'}{$host}{'clone_state'} = $script_topology{$host}->{'clone_state'} }
+            if (defined $script_topology{$host}->{'score'}) { $topology{'Host'}{$host}{'score'} = $script_topology{$host}->{'score'} }
+            if (defined $script_topology{$host}->{'version'}) { $topology{'Host'}{$host}{'version'} = $script_topology{$host}->{'version'} }
+        }
+
+        # New structure of key 'Global' with renamed keys
+        if (defined $script_topology{'global'}->{'cib-time'}) { $topology{'Global'}{'global'}{'cib-last-written'} = $script_topology{'global'}->{'cib-time'} }
+        if (defined $script_topology{'global'}->{'maintenance'}) { $topology{'Global'}{'global'}{'maintenance-mode'} = $script_topology{'global'}->{'maintenance'} }
 
-    return \%topology;
+        # We encode to the JSON to be sure that output is always same
+        $topology_json = encode_json(\%topology);
```

Comment:
What's the purpose of calling `encode_json` and then right after calling `decode_json`? Couldn't you have returned `\%topology` directly?